src/python/pants/base/build_file.py

```diff
@@ -16,6 +16,10 @@
 logger = logging.getLogger(__name__)
 
 
+def is_build_file_name(name: str) -> bool:
+    return bool(re.match(rf"^BUILD(\.[a-zA-Z0-9_-]+)?$", name))
```

Comment:
This is configurable here: https://github.com/pantsbuild/pants/blob/5864869a97bcbe406051bd68ce0be8497178828a/src/python/pants/engine/internals/mapper.py#L175-L200 so probably shouldn't be hardcoded.

Reply:
Hm. How about I make it a private function with a `_`? This PR isn't changing the prior functionality, and I think this code path isn't used in V2. My only goal is to stop using `BuildFile`.

Reply:
Fine with that... I didn't see where it had been moved from. Would be good to reference the canonical implementation here.